GDM.HW3.ConditionalsMethods/Program.cs

```diff
@@ -0,0 +1,315 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GDM.HW3.ConditionalsMethods
+{
+    class Program
+    {
+        //Task 2
+        static double Factorial(double fac)
+        {
+            if (fac == 0)
+            {
+                return 1;
+            }
+            else
+            {
+                return fac * Factorial(fac - 1);
+            }
+        }
+
+        //Task 3
+        static int Fibonachi(int n)
+        {
+            return n > 1 ? Fibonachi(n - 1) + Fibonachi(n - 2) : n;
+        }
+
+        //Task 4
+        static int DaysNumberInMonth(int month, int year)
+        {
+            if (month == 2 && year % 4 == 0 && year % 100 != 0 || month == 2 && year % 100 == 0)
+            {
+                return 29;
+            }
+            if (month == 2)
+            {
+                return 28;
+            }
+            if (month % 2 == 1 && month < 7 || month > 8 && month % 2 == 1)
+            {
+                return 30;
+            }
+            else
+            {
+                return 31;
+            }
+        }
+
+        //Task 5
+        static int[] ReccursiveQuickArraySort(int[] array, int firstArrayIndex, int lastArrayIndex)
+        {
+            int f = firstArrayIndex;
+            int l = lastArrayIndex;
+            int mid = array[(f + l) / 2];
+            do
+            {
+                while (array[f] < mid)
+                    ++f;
+                while (array[l] > mid)
+                    --l;
+                if (f <= l)
+                {
+                    if (f < l)
+                    {
+                        int temp = array[f];
+                        array[f] = array[l];
+                        array[l] = temp;
+                    }
+                    ++f;
+                    --l;
+                }
+            }
+            while (f <= l);
+            if (firstArrayIndex < l)
+            {
+                ReccursiveQuickArraySort(array, firstArrayIndex, l);
+            }
+            if (f < lastArrayIndex)
+            {
+                ReccursiveQuickArraySort(array, f, lastArrayIndex);
+            }
+            return array;
+        }
+
+        //Task 6
+        static int[] PutZeroBetweenMimMaxValueArray(int[] array)
+        {
+            int[] tempArray = (int[])array.Clone();
+            int[] sortedRandomArray = ReccursiveQuickArraySort(tempArray, 0, tempArray.Length - 1);
+            int min = 0;
+            int max = 0;
+
+            for (int i = 0; i < array.Length; i++)
```

Comment:
Why do use 2 time the same for loop on 95 + 102 lines?

Reply:
cos I don't know if the max value has bigger index than min one or not.